sdk/src/main/java/com/horizen/transaction/ForgingStakeTransaction.java

```diff
@@ -0,0 +1,256 @@
+package com.horizen.transaction;
+
+import com.google.common.primitives.Bytes;
+import com.google.common.primitives.Ints;
+import com.google.common.primitives.Longs;
+import com.horizen.box.*;
+import com.horizen.box.data.*;
+import com.horizen.proof.Proof;
+import com.horizen.proof.Signature25519;
+import com.horizen.proof.Signature25519Serializer;
+import com.horizen.proposition.Proposition;
+import com.horizen.secret.PrivateKey25519;
+import com.horizen.utils.BytesUtils;
+import com.horizen.utils.DynamicTypedSerializer;
+import com.horizen.utils.ListSerializer;
+import com.horizen.utils.Pair;
+
+import java.util.*;
+
+public final class ForgingStakeTransaction
```

Comment:
ForgingStakeTransaction is very similar to RegularTransaction. But at the moment I'm not sure that it's good to combine them together. What if spending of ForgerBoxes (or any other boxes) will require additional verification on transaction layer (so box unlocker proof itself is not enough).
Also, ForgingStakeTransaction has no Creator class, because we always need to specify what exactly ForgerBox we want to open.
Should be discussed.